### PR TITLE
email digest & debug impersonation

### PR DESCRIPTION
**Pull request by 1egoman**
screen shot 2019-03-04 at 1 22 31 pm

screen shot 2019-03-04 at 1 21 45 pm

### CODE REVIEW

File: src/components/dashboard-digest-management-modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,619 @@
+import React, { Fragment, Component } from 'react';
```
[the remark this answers is not in the document]
Comment (guscost) [reply src/components/dashboard-digest-management-modal/index.tsx]: having names that don't line up bothers me more than whether they're in the same file. two suggestions:
- common prefixes like DigestManagementRecipientListItem would be nice
- can't we put several files in the same folder? i know that would be a new thing, but how is that worse than a big file, if we have the folder already?

[the remark this answers is not in the document]
Comment (1egoman) [reply src/components/dashboard-digest-management-modal/index.tsx]: I will definitely do `common prefixes like DigestManagementRecipientListItem would be nice`. I think that I want to think about this a little more and later on break up multiple of these longer files at the same time. Thanks for the insight!

File: src/actions/route-transition/dashboard-detail.ts [hunk 2]
```diff
@@ -82,5 +85,22 @@ export default function routeTransitionDashboardDetail(id) {
...
+      schedules = await fetchAllPages(page => core.digest_schedules.list({page, page_size: 5000}));
+    } catch (err) {
+      errorThrown = err;
+    }
+    if (!errorThrown) {
+      dispatch(collectionDispatchSchedulesSet(schedules));
+    } else {
+      console.error(errorThrown);
+      dispatch(collectionDispatchSchedulesError(errorThrown));
+    }
```
[the remark this answers is not in the document]
Comment (1egoman) [reply src/actions/route-transition/dashboard-detail.ts]: Done in e12391e4

**Comment by guscost** [comment]
...except for developer tools, pending fix on that...

File: src/components/admin-token-update-modal/index.tsx [hunk 3]
```diff
@@ -73,9 +73,9 @@ export default class TokenUpdateModal extends React.Component<any, any> {
                 disabled={this.state.name.length === 0}
                 width="100%"
                 onClick={() => this.props.onSubmit({
-                  name: this.state.name,
-                  description: this.state.description,
                   key: this.state.key,
+                  name: this.state.name,
+                  description: this.state.description || undefined,
```
Comment (1egoman) [inline src/components/admin-token-update-modal/index.tsx]: What does this line accomplish?

Comment (guscost) [reply src/components/admin-token-update-modal/index.tsx]: the API doesn't accept a present-but-empty description field

**Pull request merged**